### DISCUSSION TITLE
Self hosted ente server package created + request for ente-server packaged release

### DISCUSSION THREAD

**Discussion by object1337** [ente-io/ente#1579] [Enhancements]
### Self hosted ente server package created
First of all, thanks a lot for open sourcing all the Ente components, including the server part! :sunglasses: 

To finally give back something to the community after year of using several Linux distributions, and since you shared your own hard work as well, I decided to the share my custom AUR package and even created an accompanying Arch Wiki page.

The AUR package URL: <https://aur.archlinux.org/packages/ente-server-git>
The Arch Linux Wiki URL: <https://wiki.archlinux.org/title/Ente_server>

Note that this is my first AUR package and Arch Linux Wiki page, so it might not be the best you ever see or read :smile:

### Request for ente-server packaged release
As you might notice, the source is currently the full Ente git repository, since there is no Ente server release (or Museum as you also call it in some of your documentation).

What about creating a ente-server packaged release, just like the ente-photos and ente-cli releases? Even just a tgz or zip file archive, with just the server directory would be awesome.

Not sure if you will have time for this any time soon (seeing the abundance of issues/requests within Github), but even if this is on your roadmap, that would already be nice to know.

Thanks in advance!

**Comment by mnvr** [comment]
It is in the wishlist for us too! Not exactly in the form that you mentioned (while cool, it terms of usefulness a tgz dump of the server directory won't be terribly useful), but in terms of intent there is an item on our radar to create a new version of our server Docker image that is "self sufficient".

* Right now, we already publish a server image - https://github.com/ente-io/ente/pkgs/container/server - so if someone already wants to run just the server without cloning the repository, they can do it.

* We'd like to extend this by providing a more self-contained image. Such an image will also have a bundled Ente web-app, DB, object storage and all pre-configured to talk to each other. Maybe also provide a external port so that mobile apps can connect to the API server. I can't say when we'll get around to doing this though (just mentioning since you were curious as to possible future directions). For more around this, see https://github.com/ente-io/ente/issues/983#issuecomment-2054808064

Hope this information was useful! Cheers.

**Reply by object1337** [reply]
Good to know you also wish to create an Ente server package!

However although I do understand that you're trying to make it an easy one-in all image or docker container, so pretty much anyone can easily set it up, I'm thinking more of traditional package, that just provides the Ente server part. My reasoning behind this is twofold:

- This allows a sysadmin to configure their own PostgreSQL, MinIO and Nginx services to be used in combination with Ente server, especially when not all of them are hosted on the same machine.
- This minimizes the effort the Ente developers (like yourself) have to take, to keep third party components up to date in your Ente server image.

My AUR package tries to achieve this as well, but as mentioned, currently requiring (a full) git repo clone.

A released package would have the benefit of knowing that the Ente developers decided that everything in that package is ready for release, and decreases the chance that sysadmin install Ente server in a broken state, due to some developers issue left in the git repo at a sudden point in time. This was also my point regarding the tgz/zip archive.

Hopefully my point of view is food for thought as well :smile: 

Anyway, thanks for your insights regarding your view of the direction the Ente server is most likely going, and again thanks for sharing!

Cheers

**Reply by mnvr** [reply]
Okay, I now get what you're saying. I don't think it is something which is on the radar but after your explanation I can see how this'll be useful. So I'll move this back to feature requests, let us see if we get more traction.

Note that we don't have the concept of LTS releases, the closest current analogue is cloning from the same hash as where the last Docker image was published from. However, while it is not a guarantee, generally the server code on main is stable ("ready for deployment to production").

Thanks for sharing the use case.

**Reply by object1337** [reply]
Thanks for the in-depth reply!
If you do get to it (make this sort of release) let me know :)

That's actually a pretty neat idea! I've adopted that idea, and do now perform a sparse checkout of the the commit that matches the latest docker release.

The sparse git checkout saves a lot if time when building the package, since I only really retrieve the server data while building the package (and the LICENSE file separately), and only checking out that specific "latest-docker" release results into a more stable release.

Thanks again!
